[FE]그룹 생성 페이지 - 게이지 바, 버튼

ChanYangYu opened the pull request:
## 관련 이슈

- closes (#13)
- closes (#15)

## 셀프 체크리스트

> 최소 1명 이상의 assign을 받아야만 Merge 가능합니다.

- [X] Warning Message 발생 유/무
- [X] Coding Convention 준수 유/무

## 작업 내용

> 구현 및 작업 내역

- [x] 게이지바 레이아웃 구현
- [x] 입력을 완료한 횟수만큼 게이지 증가 구현
- [x] 게이지 증가, 감소 애니메이션
- [x] 버튼 배치 및 디자인
- [x] 버튼 클릭시 상단 게이지바 감소 처리 구현
- [x] 버튼 클릭시 상단 게이지바 증가 처리 구현


## Preview

> 선택 사항

![image](https://user-images.githubusercontent.com/55623688/140011357-e06f710f-cb1b-48f2-8962-21ddfd254ced.png)

## PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- 상위 컴포넌트인 `CreateGroupPage`의 state인`ContentsNumber`를 props로 입력받아 동적 게이지 Percentage를 계산하였습니다.
- 버튼 컴포넌트도 이슈로 쪼개기에는 양이 얼마 안되서 같이 올렸습니다!
- 버튼 컴포넌트 재사용성에 대해서 고민을 많이해봤는데 아직 좋은방법이 떠오르지 않네요... 버튼 컴포넌트 위주로 리뷰를 해주시면 좋을것같습니다!

ChanYangYu committed:
[FE] design: gageBar 기본 레이아웃 구현 


- 게이지바  디자인
  - Container, ProgressBar, Percentage 스타일 적용및 배치
- issue (#13)

ChanYangYu committed:
[FE] feat: GageBar State 로직 추가 


- CreateGroupPage의 state인 contentsNumber를 props로 받아 현재 페이지 Percentage  계산
- 계산한 Page에 따라 동적으로 게이지 변경되도록 style 구현
- 게이지바 애니메이션 추가 => transition 사용

Donggggg reviewed (approved):
해당 PR에는 보이지 않는데 
#36 PR에서 작성된 clickNextContents 함수에서 4를 상수화하는 것은 어떨까요?
```
const clickNextContents = () =>{
    if(contentsNumber < 4)
      setContentsNumber(contentsNumber+1);
  }
```

ChanYangYu committed:
# [FE] refactor: 하드코드 숫자 상수화 


- ContentIndex Max값 상수화 처리
- CreateForm 마진값 변경

ChanYangYu committed:
[FE] feat: CustomButton 컴포넌트 구현

- 그룹생성 페이지 하단에 사용될 버튼 컴포넌트 구현
- 재사용을 고려해 라벨, 색상, 클릭이벤트를 props로 받아 동작하도록 구현
- issue (#15)

ChanYangYu committed:
[FE] feat: CustomButton 컴포넌트 적용


- CreateGroupPage의 button -> CustomButton 적용
- issue (#15)

yuchanleeme commented on client/src/pages/CreateGroupPage/index.tsx:
```diff
@@ -27,30 +30,40 @@ function CreateGroupPage(): JSX.Element {
   
   const clickNextContents = () =>{
-    if(contentsNumber < 4)
+    if(contentsNumber < MAX_CONTENT_INDEX)
```
const로 분리한것 좋은것 같습니다! 해당 사항도 저희 네이밍 컨벤션 wiki에 적어놓겠습니다.

ChanYangYu replied:
넵넵!

yuchanleeme reviewed (approved):
전체적으로 깔끔하게 구현하셨네요~~!! 수고하셨습니다!!
